Packs/IOCParser/Integrations/IOCParser/IOCParser.py

```diff
@@ -223,57 +228,53 @@ def unite_all_tweets_into_dict(twitter_response: Dict[str, Any]) -> None:
 
 def test_module(client: Client) -> str:
     """
-    :type client: ``Client``
-    :param client: IOCParser client to use
-
+    Tests API connectivity.
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Args:
+        client: IOCParser client to use
 
-    :return: 'ok' if test passed, anything else will fail the test.
-    :rtype: ``str``
+    Returns:
+        'ok' if test passed, anything else will fail the test
     """
-
-    response = client.ioc_from_url('https://pastebin.com/iMzrRXbJ')
-    if (response.get('status') == 'fail') \
-            or (response.get('status') == 'error') \
-            or (response.get('status') is None):
-        return 'Failed to connect with the API'
+    try:
+        client.ioc_from_url('https://pastebin.com/iMzrRXbJ')
+    except Exception as e:
+        return f'Failed to connect with the API. Error: {str(e)}'
     return 'ok'
 
 
 def ioc_from_url_command(client: Client, args: Dict[str, Any]) -> List[CommandResults]:
     """
-    Returns the results of the Parse IOCs from URL API call
+    Returns the results of the Parse IOCs from URL API call.
     Args:
-        client: IOCParser client to use
-        args: All command arguments, ulr, limit and keys (if specified)
+        client: IOCParser client to use.
+        args: All command arguments, url, limit, keys (if specified).
 
     Returns:
         CommandResults object containing the results of the parse from url as
-        returned from the API and its readable output
-
+        returned from the API and its readable output.
     """
-    url = args.get('url')
-    keys = argToList(args.get('keys'))
-    limit = args.get('limit')
-    if not keys:
-        keys = KEYS
-    keys = list_to_upper_case(keys)
-
-    if not url:
-        raise ValueError('url not specified')
-
-    response = client.ioc_from_url(url)
 
-    response_data = process_response(response, keys, limit)
+    url = args.get('url')
+    keys = list_to_upper_case(argToList(args.get('keys'))) or KEYS
+    limit = arg_to_number(args.get('limit'))
+    try:
+        response = client.ioc_from_url(url)
+    except DemistoException as e:
+        raise ValueError(str(e))
```

Comment:
Why are you raising a DemistoException as ValueError?
Why not just let it be caught as a DemistoException outside?